packages/event-migration/src/index.ts

```diff
@@ -123,41 +124,97 @@ const originalEvents = await sourceConnection.many(
 
 const idVersionHashMap = new Map<string, number>();
 
+const { parseEventType, decodeEvent, parseId } = match(config.targetDbSchema)
+  .with("catalog", () => {
+    if (!config.sourceDbSchema.includes("catalog")) {
+      throw Error(
+        "Source and target databases are incompatible, please double-check the config"
+      );
+    }
+    const parseEventType = (event_ser_manifest: any) =>
+      match(
+        event_ser_manifest
+          .replace("it.pagopa.interop.catalogmanagement.model.persistence.", "")
+          .split("|")[0]
+      )
+        .when(
+          (originalType) => (originalType as string).includes("CatalogItem"),
+          (originalType) =>
+            (originalType as string).replace("CatalogItem", "EService")
+        )
+        .otherwise((originalType) => originalType);
+
+    const decodeEvent = (eventType: string, event_payload: any) =>
+      EServiceEventV1.safeParse({
+        type: eventType,
+        event_version: 1,
+        data: event_payload,
+      });
+
+    const parseId = (anyPayload: any) =>
+      anyPayload.eservice ? anyPayload.eservice.id : anyPayload.eserviceId;
+
+    return { parseEventType, decodeEvent, parseId };
+  })
+  .with("attribute", () => {
+    if (!config.sourceDbSchema.includes("attribute")) {
+      throw Error(
+        "Source and target databases are incompatible, please double-check the config"
+      );
+    }
+    const parseEventType = (event_ser_manifest: any) =>
+      match(
+        event_ser_manifest
+          .replace(
+            "it.pagopa.interop.attributeregistrymanagement.model.persistence.",
+            ""
+          )
+          .split("|")[0]
+      )
+        .when(
+          (originalType) =>
+            (originalType as string).includes("AttributeDeleted"),
+          (originalType) =>
+            (originalType as string).replace(
+              "AttributeDeleted",
+              "MaintenanceAttributeDeleted"
+            )
+        )
+        .otherwise((originalType) => originalType);
+
+    const decodeEvent = (eventType: string, event_payload: any) =>
+      AttributeEvent.safeParse({
+        type: eventType,
+        event_version: 1,
+        data: event_payload,
+      });
+
+    const parseId = (anyPayload: any) =>
+      anyPayload.attribute ? anyPayload.attribute.id : anyPayload.id;
+
+    return { parseEventType, decodeEvent, parseId };
+  })
+  .exhaustive();
```

Comment:
Based on which package we are migrating, we have to select different implementations for these functions:
- `parseEventType`: specific rules about how to map the event name
- `decodeEvent`: which parser has to be used for that class of events (e.g. `AttributeEvent` or `EServiceEventV1`)
- `parseId`: how to extract the id from the object